datasync/downloader/downloader.go

```diff
@@ -577,6 +584,108 @@ func (d *Downloader) spawnSync(fetchers []func() error, peerID string) error {
 	return err
 }
 
+func (d *Downloader) SyncStakingInfo(id string, from, to uint64) error {
+	if d.fixing == true {
+		return errors.New("already syncing")
+	}
+	logger.Info("start syncing staking infos", "from", from, "to", to)
+	d.fixing = true
```

Comment:
How about simply adding this and removing `d.fixing = false` lines in error cases? 
```go
defer func() { d.fixing = false }()
```